src/orion/benchmark/task/profet/model_utils.py

```diff
@@ -0,0 +1,483 @@
+import json
+import pickle
+from copy import deepcopy
+from dataclasses import dataclass
+from logging import getLogger as get_logger
+from pathlib import Path
+from typing import (
+    Callable,
+    ClassVar,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import GPy
+import numpy as np
+import torch
+from emukit.examples.profet.meta_benchmarks.architecture import get_default_architecture
+from emukit.examples.profet.train_meta_model import download_data
+from GPy.models import BayesianGPLVM
+from pybnn.bohamiann import Bohamiann
+from torch import nn
+
+
+logger = get_logger(__name__)
+
+# NOTE: Split up these shared properties into the different config classes.
+# NAMES: Dict[str, str] = dict(
+#     forrester="data_sobol_forrester.json",
+#     svm="data_sobol_svm.json",
+#     fcnet="data_sobol_fcnet.json",
+#     xgboost="data_sobol_xgboost.json",
+# )
+# get_architecture: Dict[str, Callable[[int], nn.Module]] = dict(
+#     forrester=get_architecture_forrester,
+#     svm=functools.partial(get_default_architecture, classification=True),
+#     fcnet=functools.partial(get_default_architecture, classification=True),
+#     xgboost=get_default_architecture,
+# )
+# hidden_space: Dict[str, int] = dict(forrester=2, fcnet=5, svm=5, xgboost=5)
+# normalize_targets: Dict[str, bool] = dict(forrester=True, fcnet=False, svm=False, xgboost=True)
+# log_cost: Dict[str, bool] = dict(forrester=False, fcnet=True, svm=True, xgboost=True)
+# log_target: Dict[str, bool] = dict(forrester=False, fcnet=False, svm=False, xgboost=True)
+
+from abc import ABC
+
+
+@dataclass
+class MetaModelConfig(ABC):
+    """Configuration options for the training of the Profet meta-model."""
```

Comment:
The attributes here should be documented in the docstring so that it shows up in sphinx doc.

Reply:
Understood. However, I'd much rather put the docstring right next to the properties, and take advantage of the dataclass format to avoid duplicating information, a bit like what's going on here: https://stackoverflow.com/questions/51125415/how-do-i-document-a-constructor-for-a-class-using-python-dataclasses

Without adding any extensions or anything, and I'm still having some issues building the docs, so my biggest priority now is to first get the docs to build properly without errors.
Then, I'll invest a tiny bit of time to try and get something like https://pypi.org/project/sphinx-autodoc-typehints/  setup, to avoid this annoying duplication of the types (in both the docstring as well as the code itself). Then, if that isn't successfull or takes more than a day to setup, I'll just duplicate everything and add the attributes/arguments to the docstring here.

Reply:
If you can manage to get the doc automatically use the typing that would be great! Let's leave this discussion open until it is resolved.

Reply:
The attributes are properly formatted in the documentation, and the `__init__` also shows up fine.